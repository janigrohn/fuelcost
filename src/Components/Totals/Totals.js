import React from 'react';
import PropTypes from 'prop-types';
import styled from "styled-components";

const Wrapper = styled.div`
	padding: 0.25rem 0.25rem 0 0.25rem;
	margin-top: auto;
	margin-bottom: 1rem;
`;

const Row = styled.div`
	display: flex;
	justify-content: space-between;
	padding: 1rem;
	font-size: 1.5rem;
	border-top: solid 2px rgba(0, 0, 0, 0.15);
`;

const Totals = ({ vehicle, route, passengers }) => {
	const getTotalPrice = () => (route.distance * vehicle.fuelConsumption / 100 * vehicle.fuelPrice).toFixed(2);
	const getPricePerPassenger = () => (getTotalPrice() / passengers).toFixed(2);
	const getPricePerKm = () => route.distance !== 0 ? (getTotalPrice() / route.distance).toFixed(2) : '0,00';
	const getNeededFuel = () => (route.distance * vehicle.fuelConsumption / 100).toFixed(1);

	return (
		<Wrapper data-testid="totals">
			<Row>
				<span>Total cost: </span>
				<span>{getTotalPrice().replace('.', ',')}€</span>
			</Row>
			<Row>
				<span>Cost / person: </span>
				<span>{getPricePerPassenger().replace('.', ',')}€</span>
			</Row>
			<Row>
				<span>Cost / km: </span>
				<span>{getPricePerKm().replace('.', ',')}€</span>
			</Row>
			<Row>
				<span>Fuel needed: </span>
				<span>{getNeededFuel().replace('.', ',')}L</span>
			</Row>
		</Wrapper>
	);
}

Totals.propTypes = {
	vehicle: PropTypes.object.isRequired,
	route: PropTypes.object.isRequired,
	passengers: PropTypes.number.isRequired
}


export default Totals;
