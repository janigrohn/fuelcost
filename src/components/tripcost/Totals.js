import React from 'react';
import styled from "styled-components";
import { useStateValue } from '../../state/state';

const Wrapper = styled.div`
	padding: 0 0.25rem;
	margin-top: auto;
`;

const Row = styled.div`
	display: flex;
	justify-content: space-between;
	padding: 1rem;
	font-size: 1.5rem;
	border-top: solid 2px rgba(0, 0, 0, 0.15);
`;

const Totals = () => {
	const [{ vehicle, route, prices, passengers }] = useStateValue();

	const getTotalPrice = () => (route.distance * vehicle.fuelConsumption / 100 * prices[vehicle.fuelType]).toFixed(2);

	const getPricePerPassenger = () => (getTotalPrice() / passengers).toFixed(2);

	const getPricePerKm = () => (getTotalPrice() / route.distance).toFixed(2);

	return (
		<Wrapper>
			<Row>
				<span>Total cost: </span>
				<span>{getTotalPrice().replace('.', ',')}€</span>
			</Row>
			<Row>
				<span>Cost / person: </span>
				<span>{getPricePerPassenger().replace('.', ',')}€</span>
			</Row>
			<Row>
				<span>Cost / km: </span>
				<span>{getPricePerKm().replace('.', ',')}€</span>
			</Row>
		</Wrapper>
	);
}

export default Totals;
