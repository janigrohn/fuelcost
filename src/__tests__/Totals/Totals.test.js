import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Totals from '../../Components/Totals/Totals';

afterEach(cleanup);

it('matches snapshot with zero totals fi', () => {
	const vehicle = {
		fuelPrice: 0,
		fuelConsumption: 0,
	};
	const route = {
		distance: 0,
	};
	const passengers = 1;

	const { container } = render(<Totals vehicle={vehicle} route={route} passengers={passengers} lang="fi" />);
	expect(container).toMatchSnapshot();
})

it('matches snapshot with zero totals en', () => {
	const vehicle = {
		fuelPrice: 0,
		fuelConsumption: 0,
	};
	const route = {
		distance: 0,
	};
	const passengers = 1;

	const { container } = render(<Totals vehicle={vehicle} route={route} passengers={passengers} lang="en" />);
	expect(container).toMatchSnapshot();
})

it('matches snapshot with calculated values fi', () => {
	const vehicle = {
		fuelPrice: 1.55,
		fuelConsumption: 6.1,
	};
	const route = {
		distance: 440,
	};
	const passengers = 9;

	const { container } = render(<Totals vehicle={vehicle} route={route} passengers={passengers} lang="fi" />);
	expect(container).toMatchSnapshot();
})

it('matches snapshot with calculated values en', () => {
	const vehicle = {
		fuelPrice: 1.55,
		fuelConsumption: 6.1,
	};
	const route = {
		distance: 440,
	};
	const passengers = 9;

	const { container } = render(<Totals vehicle={vehicle} route={route} passengers={passengers} lang="en" />);
	expect(container).toMatchSnapshot();
})