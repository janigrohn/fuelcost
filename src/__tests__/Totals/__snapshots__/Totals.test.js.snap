// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with calculated values  1`] = `
<div>
  <div
    class="sc-bdVaJa hiQrrV"
    data-testid="totals"
  >
    <div
      class="sc-bwzfXH hoiPJK"
    >
      <span>
        Total cost: 
      </span>
      <span>
        41,60
        €
      </span>
    </div>
    <div
      class="sc-bwzfXH hoiPJK"
    >
      <span>
        Cost / person: 
      </span>
      <span>
        4,62
        €
      </span>
    </div>
    <div
      class="sc-bwzfXH hoiPJK"
    >
      <span>
        Cost / km: 
      </span>
      <span>
        0,09
        €
      </span>
    </div>
    <div
      class="sc-bwzfXH hoiPJK"
    >
      <span>
        Fuel needed: 
      </span>
      <span>
        26,8
        L
      </span>
    </div>
  </div>
</div>
`;

exports[`matches snapshot with zero totals 1`] = `
<div>
  <div
    class="sc-bdVaJa hiQrrV"
    data-testid="totals"
  >
    <div
      class="sc-bwzfXH hoiPJK"
    >
      <span>
        Total cost: 
      </span>
      <span>
        0,00
        €
      </span>
    </div>
    <div
      class="sc-bwzfXH hoiPJK"
    >
      <span>
        Cost / person: 
      </span>
      <span>
        0,00
        €
      </span>
    </div>
    <div
      class="sc-bwzfXH hoiPJK"
    >
      <span>
        Cost / km: 
      </span>
      <span>
        0,00
        €
      </span>
    </div>
    <div
      class="sc-bwzfXH hoiPJK"
    >
      <span>
        Fuel needed: 
      </span>
      <span>
        0,0
        L
      </span>
    </div>
  </div>
</div>
`;
